iroha/src/query.rs

```diff
@@ -40,6 +42,78 @@ impl TryFrom<SignedQueryRequest> for VerifiedQueryRequest {
     }
 }
 
+/// Unsupported version error
+#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+pub struct UnsupportedVersionError {
+    /// Version that we got
+    pub version: u8,
+}
+
+impl std::error::Error for UnsupportedVersionError {}
+
+impl std::fmt::Display for UnsupportedVersionError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Unsupported version. Expected version 1, got: {}",
+            self.version
+        )
+    }
+}
+
+/// Decode verified query error
+#[derive(Error, Debug)]
+pub enum DecodeVerifyQueryError {
+    /// Transaction has unsupported version
+    #[error("Transaction has unsupported version")]
+    UnsupportedQueryVersion(#[source] UnsupportedVersionError),
+    /// Failed to decode signed query
+    #[error("Failed to decode signed query")]
+    DecodeVersionedSignedQuery(#[source] iroha_version::error::Error),
+    /// Failed to verify query request
+    #[error("Failed to verify query request")]
+    VerifyQuery(iroha_error::Error),
+}
+impl iroha_http_server::http::HttpResponseError for DecodeVerifyQueryError {
```

Comment:
Please import `HttpResponseError`

Reply:
I probably forgot to add this to codestyle, but in general we try to import everything beforehand and use `::` only for calling top level functions in crates like `task::spawn(..)`.

There are many places like this, please, check it in all new additions.

Reply:
When there is a naming conflict import with renaming into upper camel case e.g: `use foo::bar as FooBar`